waltid-services/waltid-wallet-api/src/main/kotlin/id/walt/webwallet/usecase/issuer/IssuerUseCaseImpl.kt

```diff
@@ -49,12 +49,13 @@ class IssuerUseCaseImpl(
         }
 
     private suspend fun fetchCredentials(url: String): List<CredentialDataTransferObject> =
-        fetchConfiguration(url).jsonObject["credentials_supported"]!!.jsonArray.map {
-            CredentialDataTransferObject(id = it.jsonObject["id"]!!.jsonPrimitive.content,
-                format = it.jsonObject["format"]!!.jsonPrimitive.content,
-                types = it.jsonObject["types"]!!.jsonArray.map {
-                    it.jsonPrimitive.content
-                })
+        fetchConfiguration(url).jsonObject["credential_configurations_supported"]!!.jsonObject.entries.map { (key, value) ->
```

Comment:
use safe calls (?.) combined with an ?: operator to provide default values or handle errors

Reply:
Code updated, please check now.